crates/matrix-sdk/src/client.rs

```diff
@@ -992,10 +992,9 @@ impl Client {
                 let mut n = 0u8;
                 let mut port = 0u16;
                 let mut res = Err(IoError::new(IoErrorKind::Other, ""));
-                let mut rng = thread_rng();
```

Comment:
Is this somehow related? I don't see how it would be.

Reply:
yes it is, we can't hold onto [rand](https://docs.rs/rand/0.6.1/rand/index.html)::[rngs](https://docs.rs/rand/0.6.1/rand/rngs/index.html)::[ThreadRng](https://docs.rs/rand/0.6.1/rand/rngs/struct.ThreadRng.html) since it's `!Send`